### PR TITLE
Added validation to the document component

### PR DESCRIPTION
**Pull request by mohammedanas**
https://tools.hmcts.net/jira/browse/RDM-3480
The Document upload component does not validate is left untouched in MANDATORY mode, when its part of complex type.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 1]
```diff
@@ -43,16 +43,22 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
   ngOnInit() {
     this.initDialog();
     let document = this.caseField.value;
+    console.log('this.caseField', this.caseField);
```
Comment (rafalkalita) [inline src/shared/components/palette/document/write-document-field.component.ts]: Please remove the logging

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 2]
```diff
@@ -43,16 +43,21 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
   ngOnInit() {
     this.initDialog();
     let document = this.caseField.value;
+    if (this.caseField.display_context && this.caseField.display_context === this.MANDATORY) {
```
Comment (mario-paniccia) [inline src/shared/components/palette/document/write-document-field.component.ts]: Mohammed this solution looks ok. 
But can I ask you to make sure we follow the same process to do Mandatory field checks as we do for other fields? If you do a search on the code base for 'MANDATORY' you will see this is already used elsewhere. 
Let's just make sure we use one solution for all our fields. This makes the code more maintainable

Comment (mohammedanas) [reply src/shared/components/palette/document/write-document-field.component.ts]: Yes you are right, I tried importing that but, the whole testing and mocking was getting way too complicated with it, I can't remember exactly what it was, but that was the reason it seemed more easier to use the String constant MANDATORY, I shall revisit it and let you know

Comment (mario-paniccia) [reply src/shared/components/palette/document/write-document-field.component.ts]: yes please @mohammedanas

**Pull request closed without merging**